src/cider/type/plan/Expr.h

```diff
@@ -160,6 +162,7 @@ class Expr : public std::enable_shared_from_this<Expr> {
  public:
   // change this to pure virtual method after all subclasses support codegen.
   virtual JITExprValue& codegen(JITFunction& func);
+  virtual JITExprValue& codegen(JITFunction& func, context::CodegenContext& context);
```

Comment:
I think ```JITFunction& func``` is unnecessary, because ```context::CodegenContext& context``` contains it. 

BTW, ```JITExprValue& codegen(JITFunction& func)``` comes from initial demo, ```CodegenContext``` didn't exist at that time. I think this interface should be replaced with ```JITExprValue& codegen(context::CodegenContext& context)``` gradually. Any thoughts? @yma11 @jikunshang

Reply:
If we use `JITExprValue& codegen(context::CodegenContext& context)`, then we need to get the func by `auto func = context.getJITFunction();`  at each expr codegen method, acceptable?

Reply:
Agree. Better to use ```JITFunction& func = *context.getJITFunction();```, so you needn't to change the previous code.